keel-sql/src/main/resources/db/changelog/20191108-environment-artifact-constraint.yml

```diff
@@ -0,0 +1,77 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-environment-artifact-constraint-table
+      author: asher
+      changes:
+        - createTable:
+            tableName: environment_artifact_constraint
+            columns:
+              - column:
+                  name: environment_uid
+                  type: char(26)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: artifact_version
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: constraint_type
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: created_at
```

Comment:
What is this column for?

Reply:
It seems more like "evaluated at" if it's recording whether or not a constraint is satisfied?

Reply:
It's just a row insertion timestamp, since this table doesn't use a ULID primary key, so that timeouts can be calculated on constraints with external judgement dependencies. A `PENDING` manual-judgement constraint on an environment will eventually surface in the UI as a notification or request for user interaction. If it's ignored, it needs to time out after a while, i.e. if `judged_at == null && created_at < now() - 2days`, we may want to treat it as `FAILED`.

Reply:
Gotcha! That makes sense.